hpbandster/utils.py

```diff
@@ -11,11 +11,30 @@
 
 
 
-def nic_name_to_host(nic_name):
-	""" translates the name of a network card into a valid host name"""
-	from netifaces import ifaddresses, AF_INET
-	host = ifaddresses(nic_name).setdefault(AF_INET, [{'addr': 'No IP addr'}] )[0]['addr']
-	return(host)
+def nic_name_to_host(nic_name=None):
+    """ translates the name of a network card into a valid host name"""
+    from netifaces import ifaddresses, AF_INET
+
+    def get_nic_name_from_system():
+        import re
+        import subprocess
+        process = subprocess.Popen("ip route get 8.8.8.8".split(),
```

Comment:
Does this assume that the node has internet access? And it's not really portable as it requires `ip` to exist.
And pinging a Google DNS is not the most reliable way to figure out the network interface.